docs/proposals/edi.md

```diff
@@ -0,0 +1,259 @@
+# Proposal: EDI
+
+_Owners_: @chathurace @ruks  
+_Reviewers_:     
+_Created_: 2023/03/31  
+_Updated_:   
+_Issue_: [#495](https://github.com/ballerina-platform/ballerina-extended-library/issues/495)
+
+## Summary
+
+To enable EDI integrations, having EDI file processing capabilities is essential. This library provides functionalities
+to parse EDI files and write additional logic on those data.
+
+## Module Overview
+
+EDI module provides functionality to read EDI files and map those to Ballerina records or 'json' type. Mappings for EDI
+files have to be provided in json format. Once a mapping is provided, EDI module can generate Ballerina records to hold
+data in any EDI file represented by that mapping. Then the module can read EDI files (in text format) in to generated
+Ballerina records or as json values, which can be accessed from Ballerina code.
+
+## EDI file structure
+
+Here is a simple sample for EDI file
+
+```
+HDR*ORDER_1201*ABC_Store*2008-01-01~
+ITM*A-250*12
+ITM*A-45*100
+ITM*D-10*58
+ITM*K-80*250
+ITM*T-46*28
+```
+
+An EDI file is a binary computer file that contains data arranged in units called data elements. Data elements are
+separated by element terminators(*), and a group of data elements makes up a data segment. Data segments are separated
+by
+segment terminators(~).
+
+## EDI mapping file
+
+In order to parse an EDI document to ballerina readable code, an EDI mapping file is required. The mapping file can be
+manually implemented by a human for a given EDI document or can be converted from an existing definition like `Smooks`
+or `ESL`.
+
+Here is a sample EDI mapping file(edi-mapping1.json) for the above EDI document.
```

Comment:
```suggestion
The below is a sample EDI mapping file (i.e., `edi-mapping1.json`) for the above EDI document.
```
Shall we have a more meaningful name for the file to reflect the functionality of the sample instead of saying `edi-mapping1`?